frontend/src/components/SearchBox/SearchBox.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { PropTypes } from 'react';
+import debounce from 'utils/debounce';
+import './SearchBox.scss';
+
+/**
+ * A search field with next and previou buttons
+ */
+export default class SearchBox extends React.Component {
+    static propTypes = {
+        // The lines of text to search.
+        rows: PropTypes.array.isRequired,
+        // Callback ith the new searchText and currentSearchRow.
+        onSearchChanged: PropTypes.func.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            // Search text to highlight.
+            searchText: '',
+            // Index of the row with the current search result, or -1 if not found.
+            currentSearchRow: -1,
+            // Whether to display the "Not found" message.
+            notFound: false,
+        };
+        this.onSearchTextChangedDebounced = debounce(this.onSearchTextChangedDebounced, 1000);
+    }
+
+    render() {
+        const { searchText, notFound } = this.state;
+        return (
+            <div className='search-box'>
+                Search:
+                <input type='text' value={searchText} onChange={this.onSearchTextChanged} />
+                <button onClick={this.onClickPrev}>Prev</button>
+                <button onClick={this.onClickNext}>Next</button>
+                { notFound && <span className="search-not-found">Not found</span> }
+            </div>
+        );
+    }
+
+    /**
+     * Non-debounced text change handler.
+     */
+    onSearchTextChanged = (event) => {
+        this.setState({searchText: event.target.value, notFound: false});
```

Comment:
Set searchText in the non-debounced handler. Otherwise, you start typing and your text appears 1 second later - annoying.